File: frontend/src/types/interview.ts
import type { Candidate } from "./candidate";
import type { PanelMember } from "./panel";

export interface Interview {
  id: number;
  candidate_id: number;
  panel_id?: number;
  round_type: string;
  scheduled_at?: string;
  status: string;
  outcome?: string;
  feedback_notes?: string;
  candidate?: Candidate;
  panel_member?: PanelMember;
}

File: backend/app/schemas/interview.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from app.schemas.candidate import Candidate
from app.schemas.panel import PanelMember

class InterviewBase(BaseModel):
    candidate_id: int
    panel_id: Optional[int] = None
    round_type: str
    scheduled_at: Optional[datetime] = None
    status: str = "Scheduled"
    outcome: Optional[str] = None
    feedback_notes: Optional[str] = None

class InterviewCreate(BaseModel):
    candidate_id: int
    panel_id: Optional[int] = None
    round_type: str
    scheduled_at: Optional[datetime] = None

class InterviewUpdate(BaseModel):
    panel_id: Optional[int] = None
    round_type: Optional[str] = None
    scheduled_at: Optional[datetime] = None
    status: Optional[str] = None
    outcome: Optional[str] = None
    feedback_notes: Optional[str] = None

class Interview(InterviewBase):
    id: int
    candidate: Optional[Candidate] = None
    panel_member: Optional[PanelMember] = None

    class Config:
        from_attributes = True
